File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/service/UserService.java
package com.rad.fairyteller.service;

import com.rad.fairyteller.domain.User;
import com.rad.fairyteller.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public void saveOrUpdate(final User user) {
        Optional.ofNullable(user).ifPresent(userRepository::save);
    }

    public void deleteUser(final Long id) {
        userRepository.delete(id);
    }
}

File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/service/UserService.kt
package com.rad.fairyteller.service

import com.rad.fairyteller.domain.user.User
import com.rad.fairyteller.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService (private val userRepository: UserRepository) {

    fun saveOrUpdate(user: User) {
        userRepository.save(user)
    }

    fun deleteUser(id: Long?) {
        userRepository.delete(id)
    }
}